test/gtest/pad_constant.hpp

```diff
@@ -0,0 +1,176 @@
+        }
+        else if((N != 0) && (W != 0))
+        {
+            return std::vector<size_t>({N, 1, 1, 1, W});
+        }
+        else if(N != 0)
+        {
+            return std::vector<size_t>({N, 1, 1, 1, 1});
+        }
+        else
+        {
+            std::cout << "Error Input Tensor Lengths\n" << std::endl;
+            return std::vector<size_t>({0});
+        }
+    }
+};
+
+std::vector<PadConstantTestCase> PadConstantTestConfigs()
+{
+    // clang-format off
+    return {
+        { 8,    120,  0,  0,   1},  
+        { 8,    120,  0,  0,   1},
+        { 8,    1023, 0,  0,   1},  
+        { 8,    1024, 0,  0,   768},
+        { 8,    1023, 0,  0,   1},
+        { 8,    1024, 0,  0,   768},
+        { 16,   1024, 0,  0,   768},  
+        { 16,   1024, 0,  0,   768},
+        { 48,   8,    0,  512, 512}, 
+        { 48,   8,    0,  512, 512},
+        { 16, 311,    0,  98,  512},
+        { 16, 311,    0,  98,  512}
+    };
+    // clang-format on
+}
+
+template <typename T>
+struct PadConstantTest : public ::testing::TestWithParam<PadConstantTestCase>
+{
+protected:
+    PadConstantTestCase pad_constant_config;
+    tensor<T> input;
+    tensor<T> output;
+
+    tensor<T> ref_output;
+
+    miopen::Allocator::ManageDataPtr input_dev;
+    miopen::Allocator::ManageDataPtr output_dev;
+
+    size_t padding[10];
+
+    void SetUp() override
+    {
+        auto&& handle       = get_handle();
+        pad_constant_config = GetParam();
+        auto gen_value      = [](auto...) { return prng::gen_descreet_uniform_sign<T>(1e-2, 100); };
+
+        auto in_dims = pad_constant_config.GetInput();
+        input        = tensor<T>{in_dims}.generate(gen_value);
+        input_dev    = handle.Write(input.data);
+
+        // Generate random padding
+        for(size_t& i : padding)
+        {
+            i = prng::gen_descreet_unsigned<size_t>(1, 5);
+        }
+
+        std::vector<size_t> out_dims;
+        for(size_t i = 0; i < 5; i++)
+        {
+            out_dims.push_back(in_dims[i] + 2 * padding[2 * i]);
+        }
+
+        output = tensor<T>{out_dims};
+        std::fill(output.begin(), output.end(), std::numeric_limits<T>::quiet_NaN());
+        output_dev = handle.Write(output.data);
+
+        ref_output = tensor<T>{out_dims};
+        std::fill(ref_output.begin(), ref_output.end(), std::numeric_limits<T>::quiet_NaN());
+    };
+
+    void RunTest()
+    {
+        auto&& handle = get_handle();
+        auto out_dims = output.desc.GetLengths();
+
+        float padding_value = 3.5f;
+
+        cpu_pad_constant_fwd<T>(input.data.data(),
+                                ref_output.data.data(),
+                                &input.desc,
+                                &output.desc,
+                                padding,
+                                padding_value);
+        miopenStatus_t status;
+
+        const size_t* pd;
+        hipMalloc(&pd, 10 * sizeof(size_t));
```

Comment:
For static arrays you do not need to use HIPmalloc.

Reply:
I think I forgot to change the this part as it's extra code that I didn't change to the new input type